msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -8726,15 +8892,14 @@ private DiscussionForumBean duplicateForum(Long originalForumId) {
 	 * @return the head of the discussion thread for the given msg
 	 */
 	private DiscussionMessageBean getThreadHeadForMessage(Message msg) {
-	    DiscussionMessageBean threadHead = new DiscussionMessageBean(msg, messageManager);
-        //make sure we have the thread head of depth 0
-        while(threadHead.getMessage().getInReplyTo() != null){
-            threadHead = new DiscussionMessageBean(
-                    messageManager.getMessageByIdWithAttachments(threadHead.getMessage().getInReplyTo().getId()), 
-                    messageManager);
-        }
-        
-        return threadHead;
+		Message inReplyTo = msg.getInReplyTo();
+		while (inReplyTo != null)
+		{
+			// Have to use getMessageByIdWithAttachments, or we'll get LazyInitializationExceptions
+			msg = messageManager.getMessageByIdWithAttachments(inReplyTo.getId());
+			inReplyTo = msg.getInReplyTo();
```

Comment:
If msg was null it would suggest that the previous line couldn't find a message with inReplyTo.getId(), which suggests that the reply hierarchy is broken. Messages are soft deleted though, so somebody would have had to tamper with the DB. I think it's okay to break hard in this situation, though we could stop what the user is doing and say "This message has been deleted" (but I'd argue that's out of scope)